source/validate.cpp

```diff
@@ -117,6 +119,34 @@ void DebugInstructionPass(ValidationState_t& _,
   }
 }
 
+// Parses OpExtension instruction and registers extension.
+void RegisterExtension(ValidationState_t& _,
+                       const spv_parsed_instruction_t* inst) {
+  const std::string extension_str = libspirv::GetExtensionString(inst);
+  Extension extension;
+  if (!ParseSpvExtensionFromString(extension_str, &extension)) {
+    _.diag(SPV_SUCCESS) << "Failed to parse OpExtension " << extension_str;
```

Comment:
This is ok but it might be clearer for the user to say "Found unrecognized extension ...."